### PR TITLE
feat: Add default user groups

### PR DESCRIPTION
**Pull request by denisonbarbosa**
We used to rely entirely on the groups information provided by the broker, but this is unreasonable since we have some default groups in unix systems that are "required" but only make sense in the OS context.

To prevent errors and reduce frustration from misconfiguration, we now ensure that the user will always be a part of his own group and part of the "users" group as well.

### CODE REVIEW

File: internal/brokers/broker.go [hunk 1]
```diff
@@ -378,6 +381,10 @@ func validateUserInfoAndGenerateIDs(brokerName string, rawMsg json.RawMessage) (
 		uInfo.UserInfo.Groups = append(uInfo.UserInfo.Groups, users.GroupInfo{Name: g.Name, GID: gid})
 	}
 
+	// Ensure that the user groups contain its own group and the users group.
+	defaultGroups := []users.GroupInfo{{Name: uInfo.Name, GID: &uInfo.UID}, {Name: "users", GID: nil}}
+	uInfo.UserInfo.Groups = append(defaultGroups, uInfo.UserInfo.Groups...)
```
[the remark this answers is not in the document]
Comment (denisonbarbosa) [reply internal/brokers/broker.go:386]: I think we can look at [adduser.conf](https://manpages.ubuntu.com/manpages/noble/en/man5/adduser.conf.5.html) and grab the defaults from there. I do have some concerns regarding complexity and layers of configuration, though (i.e. we could be expecting too much from the user).

[the remark this answers is not in the document]
Comment (3v1n0) [reply internal/brokers/broker.go:386]: Yeah, it seems more configuration-scenario to me too.

**Review by didrocks — CHANGES_REQUESTED** [review]
Summary from the meeting:
- only create a main group matching the user's group and add it automatically as the main group of the user.
- do not add to any other groups for now.

This gives the least amount of surprise for the directory administrator.

File: internal/brokers/broker.go [hunk 2]
```diff
@@ -378,6 +381,10 @@ func validateUserInfoAndGenerateIDs(brokerName string, rawMsg json.RawMessage) (
 		uInfo.UserInfo.Groups = append(uInfo.UserInfo.Groups, users.GroupInfo{Name: g.Name, GID: gid})
 	}
 
+	// Ensure that the user groups contain its own group and that it is the main one (first on the list).
+	defaultGroups := []users.GroupInfo{{Name: uInfo.Name, GID: &uInfo.UID}}
```
Comment (didrocks) [inline internal/brokers/broker.go:385]: it’s `defaultGroup` for now :)

**Review by didrocks — APPROVED** [review]
One minor variable name comment, but then, should be good to merge!

**Pull request merged**